plugins/userdata/EventUserData.cc

```diff
@@ -0,0 +1,182 @@
+        muonInputTag_(consumes < std::vector<pat::Muon> > (iConfig.getParameter < edm::InputTag > ("muonCollection"))),
+        jetInputTag_(consumes < std::vector<pat::Jet> > (iConfig.getParameter < edm::InputTag > ("jetCollection"))), //
+        bJetInputTag_(consumes < std::vector<pat::Jet> > (iConfig.getParameter < edm::InputTag > ("bJetCollection"))), //
+        metInputTag_(consumes < std::vector<pat::MET> > (iConfig.getParameter < edm::InputTag > ("metInputTag"))), //
+        isElectron_(iConfig.getParameter<bool>("isElectron")), //
+        prefix_(iConfig.getParameter < std::string > ("prefix")), //
+        suffix_(iConfig.getParameter < std::string > ("suffix"))
+{
+  //register your products
+  produces<double>(prefix_ + "HT" + suffix_);
+  produces<double>(prefix_ + "ST" + suffix_);
+  produces<double>(prefix_ + "M3" + suffix_);
+  produces<double>(prefix_ + "Mbl" + suffix_);
+  produces<double>(prefix_ + "anglebl" + suffix_);
+  produces<double>(prefix_ + "MT" + suffix_);
+  produces<double>(prefix_ + "WPT" + suffix_);
+}
+
+EventUserData::~EventUserData()
+{
+
+  // do anything here that needs to be done at destruction time
+  // (e.g. close files, deallocate resources etc.)
+
+}
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+void EventUserData::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+  edm::Handle < std::vector<pat::Electron> > electrons;
+  iEvent.getByToken(electronInputTag_, electrons);
+
+  edm::Handle < std::vector<pat::Muon> > muons;
+  iEvent.getByToken(muonInputTag_, muons);
+
+  edm::Handle < std::vector<pat::Jet> > jetHandle;
+  iEvent.getByToken(jetInputTag_, jetHandle);
+  const pat::JetCollection& jets = *jetHandle.product();
+
+  edm::Handle < std::vector<pat::Jet> > bJetHandle;
+  iEvent.getByToken(bJetInputTag_, bJetHandle);
+  const pat::JetCollection& bjets = *bJetHandle.product();
+
+  edm::Handle < std::vector<pat::MET> > mets;
+  iEvent.getByToken(metInputTag_, mets);
+
+  const pat::MET & met(mets->at(0));
+
+  const double DEFAULT_VALUE=-999;
+  std::auto_ptr<double> ht(new double(utils::ht(jets)));
+  std::auto_ptr<double> st(new double(DEFAULT_VALUE));
+  std::auto_ptr<double> m3(new double(DEFAULT_VALUE));
+  std::auto_ptr<double> m_bl(new double(DEFAULT_VALUE));
+  std::auto_ptr<double> angle_bl(new double(-9));
+  std::auto_ptr<double> mt(new double(DEFAULT_VALUE));
+  std::auto_ptr<double> wpt(new double(DEFAULT_VALUE));
+
+  if (jets.size() >= 3)
+    *m3 = utils::m3(jets);
+
+  if ((electrons.isValid() && isElectron_) ||  (muons.isValid() && !isElectron_)) {
+    const reco::Candidate* lepton =
+        isElectron_ ?
+            (reco::Candidate*) &electrons->front() :
+            (reco::Candidate*) &muons->front();
+    if (lepton) {
+      if (bjets.size() > 0) {
+        *m_bl = utils::m_bl(bjets, lepton);
+        *angle_bl = utils::angle_bl(bjets, lepton);
+      }
+      *st = utils::st(jets, met, lepton);
```

Comment:
I went with the definition in our slides which always includes the lepton pT. I understand that there are more than one :).